python/benchmarks/common.py

```diff
@@ -347,3 +347,15 @@ def get_type_and_builtins(self, n, type_name):
         }
         data = factories[kind](n)
         return ty, data
+
+
+# RANDS_CHARS and rands are copyright pandas, see
```

Comment:
I'm not sure this is the best or even a correct way of copying in this code from pandas. I think my preference would be to created a new file in this same directory called `pandas.py`, include the Pandas BSD license text instead of the APL-2 text, and copy all of this code in. @jorisvandenbossche @AlenkaF do either of you know if we need to do this one way or another?

Reply:
I am not sure about copying code but to me the current approach feels ok. As the amount of code is minimal the note about origin of it together with copyright looks good. Joris will know better though,

Reply:
I don't think we necessarily have to copy the exact code from pandas (and deal with the licensing stuff), instead we can write something similar from scratch?

If you look in `pyarrow/tests/util.py`, I think we have already something more or less identical

Reply:
Or exactly identical ;) You can probably just change the import from pandas to `from pyarrow.tests.util import rands`

Reply:
Ahh, yep. That worked. I should've looked first, thanks for noticing.